x/claim/keeper/claim.go

```diff
@@ -43,6 +43,30 @@ func (k Keeper) EndAirdrop(ctx sdk.Context) error {
 	return nil
 }
 
+// ClawbackAirdrop implements prop 32 by clawing back all the OSMO and IONs from airdrop
+// recipient accounts with a sequence number of 0
+func (k Keeper) ClawbackAirdrop(ctx sdk.Context) error {
+	for bechAddr := range types.AirdropAddrs {
+		addr, err := sdk.AccAddressFromBech32(bechAddr)
+		if err != nil {
+			return err
+		}
+		seq, err := k.accountKeeper.GetSequence(ctx, addr)
+		if err != nil {
+			return err
+		}
+		if seq == 0 {
+			osmoBal := k.bankKeeper.GetBalance(ctx, addr, "uosmo")
+			ionBal := k.bankKeeper.GetBalance(ctx, addr, "uion")
```

Comment:
How does it work if someone who owns this address didn't claim airdrop but have sent some of his other account balance to this account?

Reply:
Heh 😅 

I supposed they'd have all their coins clawed back, including the non-genesis ones that were sent there.  I think that's fine.  Seems like a very weird case that's not worth dealing with.